src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -256,6 +255,73 @@ internal static string GetExePath(int processId = -1)
             return Interop.Sys.ReadLink(exeFilePath);
         }
 
+        /// <summary>Gets the name that was used to start the process, or null if it could not be retrieved.</summary>
+        /// <param name="processId">The pid for the target process, or -1 for the current process.</param>
+        internal static string GetProcessName(int processId = -1)
+        {
+            string cmdLineFilePath = processId == -1 ?
+                Interop.procfs.SelfCmdLineFilePath :
+                Interop.procfs.GetCmdLinePathForProcess(processId);
+
+            byte[] rentedArray = null;
+            try
+            {
+                // bufferSize == 1 used to avoid unnecessary buffer in FileStream
+                using (var fs = new FileStream(cmdLineFilePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                {
+                    Span<byte> buffer = stackalloc byte[512];
+                    int bytesRead = 0;
+                    while (true)
+                    {
+                        // Resize buffer if it was too small.
+                        if (bytesRead == buffer.Length)
+                        {
+                            uint newLength = (uint)buffer.Length * 2;
+
+                            byte[] tmp = ArrayPool<byte>.Shared.Rent((int)newLength);
+                            buffer.CopyTo(tmp);
+                            if (rentedArray != null)
+                            {
+                                ArrayPool<byte>.Shared.Return(rentedArray);
+                            }
+                            buffer = rentedArray = tmp;
```

Comment:
Nit: I'd prefer it if the code were instead something like:
```C#
byte[] toReturn = rentedArray;
buffer = rentedArray = tmp;
if (toReturn != null)
{
    ArrayPool<byte>.Shared.Return(toReturn);
}
```
The concern is if something were to go wrong (e.g. an OOM) resulting in an exception after the rented array was stored into the pool but before rentedArray was updated to point to a different buffer, then that exception would traverse the finally block, which would return the same buffer to the pool again, resulting in the single buffer being in the pool twice.